### PR TITLE
PRA in flowbased

### PR DESCRIPTION
**Pull request by terriervik**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
Preventive remedial actions are now handled in flowbased computation. 
If the crac contains rao results, the selected RAs are given a new usage rule "enforced". Else, all RAs are given a new usage rule "enforced".
Then, all enforced RAs are applied to a clone of the network.
Eventually, the flowbased computation is launched on the clone.

### CODE REVIEW

File: flowbased-computation/flowbased-computation-impl/src/test/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImplTest.java [hunk 1]
```diff
@@ -165,4 +197,39 @@ private double getCurativePtdf(FlowbasedComputationResult result, String conting
                 ))
             ));
     }
+
+    /* MOCKED ANYWAY...
```
Comment (terriervik) [inline flowbased-computation/flowbased-computation-impl/src/test/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImplTest.java:201]: I realized that the tests were based on a mock sensi computation. So my flowbased modifications have no effect...

**Comment by terriervik** [comment]
@murgeyseb I created this new PR in replacement of the former one. In my opinion, this one can be integrated before your refactoring of April. Do you agree?

File: flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java [hunk 2]
```diff
@@ -56,6 +57,10 @@ public String getVersion() {
                 .withPtdfSensitivities(glsk, crac.getBranchCnecs(), Collections.singleton(Unit.MEGAWATT))
                 .build();
 
+        String initialNetworkId = network.getVariantManager().getWorkingVariantId();
+        network.getVariantManager().cloneVariant(initialNetworkId, "InitialStateWithPra");
+        network.getVariantManager().setWorkingVariant("InitialStateWithPra");
```
Comment (murgeyseb) [inline flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java:62]: Restore initial variant at the end of the computation

**Review by murgeyseb — CHANGES_REQUESTED** [review]

File: flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java (hunk 2 above)
Comment (terriervik) [reply flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java:62]: Done

**Review by murgeyseb — APPROVED** [review]
It seems ok for me. I'll try running our integration tests to be sure it has no impact on standard Search tree algorithm.

**Pull request merged**